pyfr/plugins/fwh.py

```diff
@@ -0,0 +1,208 @@
+from collections import namedtuple
+
+import numpy as np
+
+from pyfr.mpiutil import get_comm_rank_root, mpi
+from pyfr.nputil import npeval
+from pyfr.plugins.base import BaseSolnPlugin, init_csv, SurfaceMixin
+
+
+FWHSurfParams = namedtuple('FWHSurfParams', ('eidxs', 'm0', 'norm', 'area',
+                                             'r_tilde_vec', 'r_star_inv',
+                                             'r_star_tilde_vec'))
+
+
+class FWHPlugin(SurfaceMixin, BaseSolnPlugin):
+    name = 'fwh'
+    systems = ['ac-euler', 'ac-navier-stokes', 'euler', 'navier-stokes']
+    formulations = ['dual', 'std']
+    dimensions = [2, 3]
+
+    def __init__(self, intg, cfgsect, suffix=None, *args, **kwargs):
+        super().__init__(intg, cfgsect, suffix)
+        comm, rank, root = get_comm_rank_root()
+
+        self.elementscls = intg.system.elementscls
+
+        self.tstart = self.cfg.getfloat(cfgsect, 'tstart', 0.0)
+        self.t_last = -np.inf
+        self.dt = self.cfg.getfloat(cfgsect, 'dt')
+        self.obsv_pts = np.array(self.cfg.getliteral(self.cfgsect,
+                                                     'observer-pts'))
+        self.nobvs = len(self.obsv_pts)
+
+        # Initialise data file
+        ndims = self.ndims
+        if rank == root:
+            header = ','.join(['t', 'x', 'y', 'z'][:self.ndims + 1] + ['mag'])
+            self.outf = init_csv(self.cfg, cfgsect, header)
+            self.outf.flush()
```

Comment:
Why do we need to flush here?

Reply:
I think this was from when I ran it with a `tstart` and wanted to ensure it was doing something.